pmlc/conversion/tile_to_pxa/tests/double_dot.mlir

```diff
@@ -28,11 +28,9 @@ func @double_dot(
 // CHECK:   %2 = affine.load %arg1[%arg4, %arg6] : memref<20x30xf32>
 // CHECK:   %3 = mulf %1, %2 : f32
 // CHECK:   "pxa.reduce"(%3, %0, %arg4, %arg5, %arg6) {agg = 1 : i64, map = #map2} : (f32, memref<10x30xf32>, index, index, index) -> ()
-// CHECK: ranges = [20, 10, 30]
```

Comment:
If we aren't checking for ranges here, perhaps we should check them elsewhere with a CHECK on the parallel for itself.

Reply:
The new results look like this:

#map0 = (d0, d1) -> (d1, d0)
#map1 = (d0, d1) -> (d0, d1)
#map2 = (d0, d1, d2) -> (d1, d2)
#map3 = () -> (20, 10, 30)
#map4 = (d0, d1, d2) -> (d0, d1, d2)
#map5 = () -> (30, 10, 40)


module {
  func @double_dot(%arg0: memref<10x20xf32>, %arg1: memref<20x30xf32>, %arg2: memref<30x40xf32>, %arg3: memref<10x40xf32>) {
    %0 = alloc() : memref<10x30xf32>
    "pxa.parallel_for"() ( {
    ^bb0(%arg4: index, %arg5: index, %arg6: index):	// no predecessors
      %1 = affine.load %arg0[%arg5, %arg4] : memref<10x20xf32>
      %2 = affine.load %arg1[%arg4, %arg6] : memref<20x30xf32>
      %3 = mulf %1, %2 : f32
      "pxa.reduce"(%3, %0, %arg4, %arg5, %arg6) {agg = 1 : i64, map = #map2} : (f32, memref<10x30xf32>, index, index, index) -> ()
      "affine.terminator"() : () -> ()
    }) {ranges = #map3, transform = #map4} : () -> ()
    "pxa.parallel_for"() ( {
    ^bb0(%arg4: index, %arg5: index, %arg6: index):	// no predecessors
      %1 = affine.load %0[%arg5, %arg4] : memref<10x30xf32>
      %2 = affine.load %arg2[%arg4, %arg6] : memref<30x40xf32>
      %3 = mulf %1, %2 : f32
      "pxa.reduce"(%3, %arg3, %arg4, %arg5, %arg6) {agg = 1 : i64, map = #map2} : (f32, memref<10x40xf32>, index, index, index) -> ()
      "affine.terminator"() : () -> ()
    }) {ranges = #map5, transform = #map4} : () -> ()
    return
  }
}

Not exactly sure how to capture the maps and them compare them.  Maybe CHECK-DAG of some sort?  Or maybe looks for the maps and get their names?

Reply:
I think check for maps then check that their name is used (using regex capture probably)